### ISSUE TITLE
circinus doesn't build: vyos-1x : Depends: salt-minion but it is not installable

### ISSUE THREAD

**Issue by dutty5** [dd010101/vyos-jenkins#57]
Hi, something weird is going on, may be temporary. circinus doesn't build because of:
```
The following packages have unmet dependencies:
 vyos-1x : Depends: salt-minion but it is not installable
```

```
Something wicked happened resolving 'repo.saltproject.io:443' (-5 - No address associated with hostname)
Reading package lists... Done
```
Looks like the saltproject has a cdn problem, or shutdown.

**Comment by dd010101** [comment]
Seems like the repository has been migrated to different domain https://saltproject.io/blog/salt-project-package-repo-migration-and-guidance/. It's `packages.broadcom.com` now. The VyOS team did fix it but only in the `current` by https://github.com/vyos/vyos-build/pull/829 but they are using their own repo for some reason. We can use little "sed" to temporarily modify the [toml](https://github.com/vyos/vyos-build/blob/circinus/data/architectures/amd64.toml) before doing build and we can use the VyOS repo since it's public.

Try the https://github.com/dd010101/vyos-jenkins/commit/c823120cd03abf3bf154df34ea61e52df1dfdf07 - it should build now.

**Comment by marekm72** [comment]
"Broadcom" doesn't sound too open-source-friendly but rather very money-friendly (how long before it becomes paywalled and that may be why they moved it so quickly - XCP-NG got many new users after VMware got acquired and had large price increases) so I guess that could be another reason to better have own repo.

BTW, I'm looking into backporting a few things to forked sagitta branch - T6807 looks trivial (one-line regex change), but in T6776 (Zabbix vulnerability) I see something odd: a binary key file, not sure if it got corrupted or is that actually some other but supported file format (the other existing key files in there look normal, simple text files).

**Comment by dd010101** [comment]
They explained why here https://github.com/vyos/vyos-build/pull/828#issuecomment-2451518953, it's temporary. The new repo doesn't have the version they want.

> I see something odd: a binary key file, not sure if it got corrupted or is that actually some other but supported file format (the other existing key files in there look normal, simple text files).

You looking at GPG public key and that's bunch of bytes, the ASCII version is "armored" with encoding but it's still the same bytes. The binary format is the native form and thus supported by the GPG and thus by the APT. It's just encoding. The armored ASCII has usually .asc and the binary has usually .gpg extension.

If you want to see if the key is valid to make sure it's not corrupted then you can import or inspect the key in given file.

To just inspect:
```
gpg --list-packets --verbose /path/to/pubkey
```

or

```
cat /path/to/pubkey | gpg --with-colons --import-options show-only --import
```

To import:

```
gpg --import < /path/to/pubkey
gpg --list-keys
```

**Comment by dd010101** [comment]
The salt repository is also fixed for sagitta https://github.com/dd010101/vyos-build/commit/e68b01c205201f54dc37f238b6b17eaa6a2b26a4.

The equuleus also uses the decommissioned repository but the build works regardless and also I have no idea where I would get the salt 3003 for buster, the VyOS repo has only [3005](https://packages.vyos.net/saltproject/debian/11/amd64/3005/pool/main/s/salt/) and the broadcom has only [3006/3007](https://packages.broadcom.com/artifactory/saltproject-deb/pool/) 🤔. Well it works without it so...

**Comment by marekm72** [comment]
> You looking at GPG public key and that's bunch of bytes, the ASCII version is "armored" with encoding but it's still the same bytes. The binary format is the native form and thus supported by the GPG and thus by the APT. It's just encoding. The armored ASCII has usually .asc extension and the binary has usually .gpg.

OK, I've just checked the file is indeed read correctly by gpg, it's just that seeing a binary file in a directory with otherwise text files raised a red flag.  The binary file contains a subset of packets from https://repo.zabbix.com/zabbix-official-repo.key - not sure why they haven't just included the original, other than to keep us a bit more busy ;)

**Comment by dutty5** [comment]
Hi, I find it a bit strange that none of the circinus packages got any update for about a month. 
I run `~/vyos-jenkins/new/package_builder.py circinus` and all packages are reported up to date. I built them a month ago initially, and still none changed.

Is this new method of building circinus still valid? Or we just use the orphaned code, so there is no use of building it?

**Comment by dd010101** [comment]
TL;DR
Yes, it's strange, the packages are stale for long time. 

Yes, the circinus method is valid as much as it can be - to my understanding.

No, the circinus isn't orphaned as of now, only the equuleus and sagitta is - per official statements, although in reality neither the circinus got any update since they orphaned the equuleus/sagitta, that was around a month ago. 

---

This reflects the current state of development - it's stale and not only the public side but also the LTS side - there is still no 1.4.1 as of now.

The Stream was planned to be released in [October](https://blog.vyos.io/vyos-project-august-2024-update) (that should be the point where we see updates for the circinus) yet there was no release nor news since it was announced in August. It's not surprising that the stream is on back burner because if they are behind on the LTS then they likely won't find time for the other stuff with lower priority. 

The source code for equuleus and sagitta was split where some packages did come from my repositories - because there was no other way to make it work, although majority did come directly from VyOS.

The circinus is different where everything comes from https://github.com/vyos. Thus if you get "up to date" then either there are no new commits in VyOS repositories or the checking of commits is broken but I know it's the former case.

The up to date check stores the commit of last build and on next run it analyzes the changed files of all commits since the last commit from the given vyos repo. You can rebuild everything or specific package if you want by using `--force`, see the `--help`. There isn't much point of doing because there were no changes for long time - look at the commit dates here https://github.com/vyos/vyos-build/tree/Circinus and here https://github.com/vyos/vyos-1x/tree/Circinus if these two repositories don't see any updates then likely all packages are stale.

**Comment by marekm72** [comment]
Don't worry, official statement may come 6 weeks later - circinus is probably only updated in private repo like equuleus/sagitta.
BTW, my first PR (in main README.md to make it less misleading) has been approved.  But now it seems I'm blocked on their GH - can't write even a short "thank you" comment in my own PR, or fork any of their repos (any random GH account can do it, but I can't) to make more new PRs that they wanted so much to give free subscriptions in return.  And silenced on their forum too, this time for "creating dissent in the community" - or did they mean "communist party"?  "Those who wanted free beer, now need accept it’s taste" (c) syncer - yeah, now with more energy from free Po 210 included! :)

**Comment by dutty5** [comment]
> circinus is probably only updated in private repo

Yes, likely. Stream may be canceled. Only rolling images will be available. Alternatively, Stream will have another public repo, and it's different from the current circinus repo which is no longer updated. Also, Stream (if released finally) may not have public repos at all. It may also have a limited functionality. One way or other, their direction is clear: there will be no more images available for normal free use.

**Comment by dd010101** [comment]
> circinus is probably only updated in private repo like equuleus/sagitta

Yes, that's what they did say about the Stream. It should be privately developed and the source will be released once given version is released. To be clear - if they give us tarballs - that's not a deal breaker. We can easily modify the scripts to build from tarballs instead of repositories, also we can just make our own repositories and update them with whatever they give us. 

I have no idea how it will work but they promised it would happen somehow, so it's reasonable to expect something.

> But now it seems I'm blocked on their GH...  And silenced on their forum too...

Are you even surprised?! If you talk mainly about something they don't want to see, you poke the thing over and over - then you easily outweigh your usefulness.

---

> Yes, likely. Stream may be canceled. Only rolling images will be available. Alternatively, Stream will have another public repo, and it's different from the current circinus repo which is no longer updated. Also, Stream (if released finally) may not have public repos at all. It may also have a limited functionality. One way or other, their direction is clear: there will be no more images available for normal free use.

I assume this is pure speculation. Let's just say I will wait and maybe something will happen and if it doesn't then it won't change a thing.

**Comment by dd010101** [comment]
They approved the first PR because it just made correction about the OpenCollective. The second PR wasn't accepted since it mentioned the officially unannounced "private development" and that's also what was mentioned in your "hijack" on the forum.

> You were told many times you can’t just go to each topic and start putting your random stuff there

As far as I know you were told only once and the final "hijack" wasn't even hijack since you only responded to the topic or reply - very nice reasoning 😄 and so the thread is hidden now...

They seem to agree with you on these changes but you were stealing their announcement, that was the issue.

**Comment by dutty5** [comment]
> something will happen and if it doesn't then it won't change a thing.

One thing that changed is that a month ago there was an idea that we can move to circinus that is going to be up to date. Currently, there is no difference between the three, equuleus, sagitta, circinus: all of them can be built however only the underlying OS is updated, not VyOS specific packages.

**Comment by dd010101** [comment]
The equuleus and sagitta are officially stated as publicly abandoned and that's backed by reasons. The circinus has still future in the Stream form, maybe it won't be called "circinus" but it would be circinus equivalent, Stream can be circinus or fork of circinus, either way it's the same outcome. Currently everything is stale including the LTS, thus the current state doesn't reflect the future state, it's just temporary, the equuleus and sagitta situation is permanent though, that's why the idea to move to circinus is still valid until proven otherwise.

**Comment by marekm72** [comment]
Looking at the future 1.3.9 and 1.4.1 changelogs, equuleus will probably see very few changes, which means we should be able to maintain it in parallel very close to the original.  Currently just T1311 and T5936, not critical really.  Interestingly, there seems to be no auto-close of PRs for equuleus and there is even PR for T1311 already (one-line fix), even if they don't accept it we could include it in the unofficial build.  T5936 seems difficult to backport, maybe it will not be after all.  On the other hand, sagitta has a lot of changes between 1.4.0 and 1.4.1 - I haven't yet looked in detail which of the listed changes are in the public branch and which will need to be backported.

**Comment by dd010101** [comment]
What changelog you are using?

This one? 
https://docs.vyos.io/en/latest/changelog/1.4.html
https://github.com/vyos/vyos-documentation/blob/current/docs/changelog/1.4.rst

T6505 https://github.com/vyos/vyos-1x/pull/3882
T5878 https://github.com/vyos/vyos-1x/pull/3759
T5949 https://github.com/vyos/vyos-1x/pull/3709
T6320 https://github.com/vyos/vyos-1x/pull/3847
T6423 https://github.com/vyos/vyos-1x/pull/3603
T6424 https://github.com/vyos/vyos-1x/pull/3619
T6454 https://github.com/vyos/vyos-1x/pull/3605
T6462 https://github.com/vyos/vyos-1x/pull/3633
T6473 https://github.com/vyos/vyos-1x/pull/3640
T6477 https://github.com/vyos/vyos-1x/pull/3738
T6538 https://github.com/vyos/vyos-1x/pull/3755
T6539 https://github.com/vyos/vyos-1x/pull/3953
T6539 https://github.com/vyos/vyos-1x/pull/3913
T6566 https://github.com/vyos/vyos-1x/pull/3807
T6614 https://github.com/vyos/vyos-1x/pull/3893
T4287 https://github.com/vyos/vyos-1x/pull/3780
T5514 https://github.com/vyos/vyos-1x/pull/3695
T5552 https://github.com/vyos/vyos-1x/pull/3862
T6148 https://github.com/vyos/vyos-1x/pull/3918
T6148 https://github.com/vyos/vyos-1x/pull/3909
T6332 https://github.com/vyos/vyos-1x/pull/3538
T6401 https://github.com/vyos/vyos-1x/pull/3602
T6429 https://github.com/vyos/vyos-1x/pull/3586
T6431 https://github.com/vyos/vyos-1x/pull/3583
T6453 https://github.com/vyos/vyos-1x/pull/3604
T6460 https://github.com/vyos/vyos-1x/pull/3600
T6463 https://github.com/vyos/vyos-1x/pull/3618
T6464 https://github.com/vyos/vyos-1x/pull/3617
T6480 https://github.com/vyos/vyos-1x/pull/3651
T6484 https://github.com/vyos/vyos-build/pull/659
T6503 https://github.com/vyos/vyos-1x/pull/3704
T6519 https://github.com/vyos/vyos-1x/pull/3725
T6523 https://github.com/vyos/vyos-1x/pull/3826
T6559 https://github.com/vyos/vyos-1x/pull/4017
T6559 https://github.com/vyos/vyos-1x/pull/3878
T6559 https://github.com/vyos/vyos-1x/pull/3824
T6584 https://github.com/vyos/vyos-build/pull/701
T6593 https://github.com/vyos/vyos-1x/pull/3866
T6600 https://github.com/vyos/vyos-1x/pull/3858
T6602 https://github.com/vyos/vyos-1x/pull/3869
T6603 https://github.com/vyos/vyos-1x/pull/3905
T6605 https://github.com/vyos/vyos-1x/pull/3878
T6610 https://github.com/vyos/vyos-build/pull/717
T4026 https://github.com/vyos/vyos-1x/pull/3667
T6290 https://github.com/vyos/vyos-build/pull/687
T6379 https://github.com/vyos/vyos-1x/pull/3926
T6446 https://github.com/vyos/vyos-build/pull/650
T6486 https://github.com/vyos/vyos-1x/pull/3926
T6500 https://github.com/vyos/vyos-1x/pull/3689
T6524 https://github.com/vyos/vyatta-cfg-system/pull/226
T6524 https://github.com/vyos/vyatta-op/pull/94
T6524 https://github.com/vyos/vyos-1x/pull/3741
T6592 https://github.com/vyos/vyos-1x/pull/3914
T6592 https://github.com/vyos/vyos-1x/pull/3837
T6592 https://github.com/vyos/vyos-1x/pull/3872
T6594 https://github.com/vyos/vyos-1x/pull/3855
T6597 https://github.com/vyos/vyos-build/pull/706
T6597 https://github.com/vyos/vyos-1x/pull/3845
T6598 https://github.com/vyos/vyos-build/pull/731
T6598 https://github.com/vyos/vyos-build/pull/721
T6598 https://github.com/vyos/vyos-1x/pull/3886
T6598 https://github.com/vyos/vyos-build/pull/719

That's all the tasks of 1.4.1 from the changelog and their associated pull-request for sagitta branch. Thus I conclude the changelog is outdated because all the changes are from the public LTS era. This is confirmed by the last update of this changelog - it was 3 months ago.

**Comment by dutty5** [comment]
> all the changes are from the public LTS era

Does it mean that those changes are included in our jenkins build if built today?

**Comment by dd010101** [comment]
Yes, all those changes are 3+ months old, thus they are already in the frozen sagitta branch. That's missing the point tho - this tells you that the changelog is not correct - we don't see what is missing because the changelog wasn't updated for long time. Maybe they will update the changelog once new version released, who knows.

**Comment by dutty5** [comment]
[November Update](https://blog.vyos.io/vyos-project-november-2024-update) suggests soon Stream images release together with the source tarballs. The new build system is probably totally hidden in the private repositories.

**Comment by dd010101** [comment]
Yep, the packaging part of the build system isn't public. If you look at `current`, you can only see the compilation part, although that's the only part we need anyway. This means that the development will be somewhat obscured, as we will not have a detailed history of the source, but only a snapshot - but not really, because many changes are parallel with the `current` and also most changes are self-explanatory.

I have no idea what tarball(s) we will get, hopefully it will be structured in sensible way (tarball or directory per repo/package) but who knows... If it's sensibly structured then it would be easy to integrate to scripts we have now. If they supply something usable then there are two routes we can take 1) commit contents of tarballs into repositories and build from git or 2) directly build from tarballs - what is better?

**Comment by dutty5** [comment]
To me, commit contents of tarballs into repositories looks more sustainable. However it depends what they release

**Comment by dutty5** [comment]
I tried now to refresh the ISO of the Sagitta. It doesn't build. [   Failed!   ] Package: vyos-xe-guest-utilities - Branch: sagitta 

Is it a known problem? I looked into the Jenkins Console output of the package and couldn't figure out the root cause. The first line that looks suspicious is this:
```
...
ERROR: The container started but didn't run the expected command.
```

**Comment by dd010101** [comment]
I don't know about any issues with equuleus/sagitta and I did test the build week ago and it was fine. I tried to build the vyos-xe-guest-utilities/sagitta now and this also works.

> ERROR: The container started but didn't run the expected command.

That's not a real error - this errors happens every time with every build - it's just side effect how the docker is invoked from the Jenkins pipeline - the entrypoint is not being used.

Full Jenkins console log would be helpful to see what exactly happened and what is the state of your Jenkins.

> Edit: More precisely I don't know if it builds, I just stopped at running 7-build-project-jobs.sh that failed with that package build.

The build would likely pass if you have previous vyos-xe-guest-utilities deb so that's not good way to test.

**Comment by dutty5** [comment]
Last time I successfully built Sagitta ISO was about 3 weeks ago. Yesterday I followed my usual workflow to build a new ISO. First I updated the dd010101/vyos-jenkins (there are several changes as far as I can tell looking to the latest commits), then ran everything starting from ./1-...sh till ./7-...sh.

./6- strep updated a number of packages, so for the long time I got new versions of packages in my Vyos Saggita repository. All of those builds were successful. Step ./7- started the build of the vyos-xe-guest-utilities package that eventually failed. I tried manually to run that build from Jenkins, it fails every time. I do have the previous version of this package in my repo, however I did not try to build the ISO. I wanted to resolve the issue with the vyos-xe-guest-utilities first. Besides, it looks like the number of packages updated by ./6-.sh step may be related to the new version of vyos-xe-guest-utilities.

I looked through the Console output of the failed build and I couldn't figure out what's wrong there. I too was doubtful that the ERROR I cited above was relevant.

Interestingly, the step 6-..sh produced the successful build of the vyos-xe-guest-utilities however without artifacts, and there is no new .deb version in the repository. Next step 7-..sh initiated the build again, which failed.

Below is the Console output of the successful build of vyos-xe-guest-utilities which produced no artifacts (started by 6-.sh), and next is the Console output of the failed build (started by 7-.sh).

<details>
 <summary>Console Output</summary>

  ```
Branch indexing
 > git rev-parse --resolve-git-dir /var/lib/jenkins/caches/git-ac5b0e7fe49407a1479ecc051169b141/.git # timeout=10
Setting origin to https://github.com/notvyos/vyos-xe-guest-utilities.git
 > git config remote.origin.url https://github.com/notvyos/vyos-xe-guest-utilities.git # timeout=10
Fetching origin...
Fetching upstream changes from origin
 > git --version # timeout=10
 > git --version # 'git version 2.39.5'
 > git config --get remote.origin.url # timeout=10
 > git fetch --tags --force --progress -- origin +refs/heads/*:refs/remotes/origin/* # timeout=10
Seen branch in repository origin/equuleus
Seen branch in repository origin/sagitta
Seen 2 remote branches
Obtained Jenkinsfile from 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4
Loading library vyos-build@sagitta
Attempting to resolve sagitta from remote references...
 > git --version # timeout=10
 > git --version # 'git version 2.39.5'
 > git ls-remote -h -- https://github.com/dd010101/vyos-build.git # timeout=10
Found match: refs/heads/sagitta revision 1c84031f2d9384040affbe4e3386f28bae5a9b14
The recommended git tool is: git
No credentials specified
Cloning the remote Git repository
Cloning with configured refspecs honoured and without tags
Cloning repository https://github.com/dd010101/vyos-build.git
 > git init /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta@libs/3153541cf11860983c97add73dc0f9e11532148c8068c6cfe37952173ab618e9 # timeout=10
Fetching upstream changes from https://github.com/dd010101/vyos-build.git
 > git --version # timeout=10
 > git --version # 'git version 2.39.5'
 > git fetch --no-tags --force --progress -- https://github.com/dd010101/vyos-build.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/dd010101/vyos-build.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
Checking out Revision 1c84031f2d9384040affbe4e3386f28bae5a9b14 (sagitta)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 1c84031f2d9384040affbe4e3386f28bae5a9b14 # timeout=10
Commit message: "docker: added missing dependencies for vyos-xe-guest-utilities"
First time build. Skipping changelog.
[Pipeline] Start of Pipeline
[Pipeline] timeout
Timeout set to expire in 4 hr 0 min
[Pipeline] {
[Pipeline] timestamps
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Define Agent)
[Pipeline] node
23:52:55  Still waiting to schedule task
23:52:55  Waiting for next available executor
23:53:09  Running on [Jenkins](http://192.168.3.180:8080/computer/(built-in)/) in /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta
[Pipeline] {
[Pipeline] checkout
23:53:09  Selected Git installation does not exist. Using Default
23:53:09  The recommended git tool is: NONE
23:53:09  No credentials specified
23:53:09  Cloning the remote Git repository
23:53:09  Cloning repository https://github.com/notvyos/vyos-xe-guest-utilities.git
23:53:09   > git init /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta # timeout=10
23:53:09  Fetching upstream changes from https://github.com/notvyos/vyos-xe-guest-utilities.git
23:53:09   > git --version # timeout=10
23:53:09   > git --version # 'git version 2.39.5'
23:53:09   > git fetch --tags --force --progress -- https://github.com/notvyos/vyos-xe-guest-utilities.git +refs/heads/*:refs/remotes/origin/* # timeout=10
23:53:10   > git config remote.origin.url https://github.com/notvyos/vyos-xe-guest-utilities.git # timeout=10
23:53:10   > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
23:53:10  Avoid second fetch
23:53:10  Checking out Revision 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 (sagitta)
23:53:10   > git config core.sparsecheckout # timeout=10
23:53:10   > git checkout -f 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 # timeout=10
23:53:10  Commit message: "fixed jenkins library"
23:53:10  First time build. Skipping changelog.
[Pipeline] withEnv
[Pipeline] {
[Pipeline] echo
23:53:10  Warning, empty changelog. Probably because this is the first build.
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // node
Stage "Define Agent" skipped due to when conditional
[Pipeline] getContext
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Build Code)
[Pipeline] echo
23:53:11  Warning, empty changelog. Probably because this is the first build.
Stage "Build Code" skipped due to when conditional
[Pipeline] getContext
[Pipeline] parallel
[Pipeline] { (Branch: amd64)
[Pipeline] { (Branch: arm64)
[Pipeline] stage
[Pipeline] { (amd64)
[Pipeline] stage
[Pipeline] { (arm64)
Stage "amd64" skipped due to when conditional
[Pipeline] getContext
[Pipeline] }
Stage "arm64" skipped due to when conditional
[Pipeline] getContext
[Pipeline] }
[Pipeline] // stage
[Pipeline] // stage
[Pipeline] }
[Pipeline] }
[Pipeline] // parallel
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Finalize)
[Pipeline] node
23:53:26  Still waiting to schedule task
23:53:26  Waiting for next available executor
23:57:39  Running on [Jenkins](http://192.168.3.180:8080/computer/(built-in)/) in /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta
[Pipeline] {
[Pipeline] checkout
23:57:39  Selected Git installation does not exist. Using Default
23:57:39  The recommended git tool is: NONE
23:57:39  No credentials specified
23:57:39   > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/.git # timeout=10
23:57:39  Fetching changes from the remote Git repository
23:57:39   > git config remote.origin.url https://github.com/notvyos/vyos-xe-guest-utilities.git # timeout=10
23:57:39  Fetching upstream changes from https://github.com/notvyos/vyos-xe-guest-utilities.git
23:57:39   > git --version # timeout=10
23:57:39   > git --version # 'git version 2.39.5'
23:57:39   > git fetch --tags --force --progress -- https://github.com/notvyos/vyos-xe-guest-utilities.git +refs/heads/*:refs/remotes/origin/* # timeout=10
23:57:40  Checking out Revision 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 (sagitta)
23:57:40   > git config core.sparsecheckout # timeout=10
23:57:40   > git checkout -f 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 # timeout=10
23:57:40  Commit message: "fixed jenkins library"
[Pipeline] withEnv
[Pipeline] {
[Pipeline] echo
23:57:40  Warning, empty changelog. Probably because this is the first build.
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // node
Stage "Finalize" skipped due to when conditional
[Pipeline] getContext
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] }
[Pipeline] // timeout
[Pipeline] End of Pipeline
Finished: SUCCESS
 ```
</details>
<details>
<summary>Console Output</summary>

```
Started from command line by [admin](http://192.168.3.180:8080/user/admin)
 > git rev-parse --resolve-git-dir /var/lib/jenkins/caches/git-ac5b0e7fe49407a1479ecc051169b141/.git # timeout=10
Setting origin to https://github.com/notvyos/vyos-xe-guest-utilities.git
 > git config remote.origin.url https://github.com/notvyos/vyos-xe-guest-utilities.git # timeout=10
Fetching origin...
Fetching upstream changes from origin
 > git --version # timeout=10
 > git --version # 'git version 2.39.5'
 > git config --get remote.origin.url # timeout=10
 > git fetch --tags --force --progress -- origin +refs/heads/*:refs/remotes/origin/* # timeout=10
Seen branch in repository origin/equuleus
Seen branch in repository origin/sagitta
Seen 2 remote branches
Obtained Jenkinsfile from 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4
Loading library vyos-build@sagitta
Attempting to resolve sagitta from remote references...
 > git --version # timeout=10
 > git --version # 'git version 2.39.5'
 > git ls-remote -h -- https://github.com/dd010101/vyos-build.git # timeout=10
Found match: refs/heads/sagitta revision 1c84031f2d9384040affbe4e3386f28bae5a9b14
The recommended git tool is: git
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta@libs/3153541cf11860983c97add73dc0f9e11532148c8068c6cfe37952173ab618e9/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/dd010101/vyos-build.git # timeout=10
Fetching without tags
Fetching upstream changes from https://github.com/dd010101/vyos-build.git
 > git --version # timeout=10
 > git --version # 'git version 2.39.5'
 > git fetch --no-tags --force --progress -- https://github.com/dd010101/vyos-build.git +refs/heads/*:refs/remotes/origin/* # timeout=10
Checking out Revision 1c84031f2d9384040affbe4e3386f28bae5a9b14 (sagitta)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 1c84031f2d9384040affbe4e3386f28bae5a9b14 # timeout=10
Commit message: "docker: added missing dependencies for vyos-xe-guest-utilities"
 > git rev-list --no-walk 1c84031f2d9384040affbe4e3386f28bae5a9b14 # timeout=10
[Pipeline] Start of Pipeline
[Pipeline] timeout
Timeout set to expire in 4 hr 0 min
[Pipeline] {
[Pipeline] timestamps
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Define Agent)
[Pipeline] node
00:00:10  Running on [Jenkins](http://192.168.3.180:8080/computer/(built-in)/) in /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta
[Pipeline] {
[Pipeline] checkout
00:00:10  Selected Git installation does not exist. Using Default
00:00:10  The recommended git tool is: NONE
00:00:10  No credentials specified
00:00:10   > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/.git # timeout=10
00:00:10  Fetching changes from the remote Git repository
00:00:10   > git config remote.origin.url https://github.com/notvyos/vyos-xe-guest-utilities.git # timeout=10
00:00:10  Fetching upstream changes from https://github.com/notvyos/vyos-xe-guest-utilities.git
00:00:10   > git --version # timeout=10
00:00:10   > git --version # 'git version 2.39.5'
00:00:10   > git fetch --tags --force --progress -- https://github.com/notvyos/vyos-xe-guest-utilities.git +refs/heads/*:refs/remotes/origin/* # timeout=10
00:00:10  Checking out Revision 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 (sagitta)
00:00:10   > git config core.sparsecheckout # timeout=10
00:00:10   > git checkout -f 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 # timeout=10
00:00:10  Commit message: "fixed jenkins library"
00:00:10   > git rev-list --no-walk 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 # timeout=10
[Pipeline] withEnv
[Pipeline] {
[Pipeline] echo
00:00:10  Warning, empty changelog. Have you run checkout?
[Pipeline] script
[Pipeline] {
[Pipeline] sh
00:00:11  + id -u
[Pipeline] sh
00:00:11  + id -g
[Pipeline] }
[Pipeline] // script
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Build Code)
[Pipeline] echo
00:00:11  Warning, empty changelog. Have you run checkout?
[Pipeline] parallel
[Pipeline] { (Branch: amd64)
[Pipeline] { (Branch: arm64)
[Pipeline] stage
[Pipeline] { (amd64)
[Pipeline] stage
[Pipeline] { (arm64)
[Pipeline] getContext
[Pipeline] node
00:00:11  Running on [Jenkins](http://192.168.3.180:8080/computer/(built-in)/) in /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta
Stage "arm64" skipped due to when conditional
[Pipeline] getContext
[Pipeline] }
[Pipeline] {
[Pipeline] // stage
[Pipeline] }
[Pipeline] checkout
00:00:11  Selected Git installation does not exist. Using Default
00:00:11  The recommended git tool is: NONE
00:00:11  No credentials specified
00:00:11   > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/.git # timeout=10
00:00:11  Fetching changes from the remote Git repository
00:00:11   > git config remote.origin.url https://github.com/notvyos/vyos-xe-guest-utilities.git # timeout=10
00:00:11  Fetching upstream changes from https://github.com/notvyos/vyos-xe-guest-utilities.git
00:00:11   > git --version # timeout=10
00:00:11   > git --version # 'git version 2.39.5'
00:00:11   > git fetch --tags --force --progress -- https://github.com/notvyos/vyos-xe-guest-utilities.git +refs/heads/*:refs/remotes/origin/* # timeout=10
00:00:12  Checking out Revision 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 (sagitta)
00:00:12   > git config core.sparsecheckout # timeout=10
00:00:12   > git checkout -f 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 # timeout=10
00:00:12  Commit message: "fixed jenkins library"
[Pipeline] withEnv
[Pipeline] {
[Pipeline] withEnv
[Pipeline] {
[Pipeline] withDockerRegistry
[Pipeline] {
[Pipeline] isUnix
[Pipeline] withEnv
[Pipeline] {
[Pipeline] sh
00:00:12  + docker pull 172.17.17.17:5000/vyos/vyos-build:sagitta
00:00:12  sagitta: Pulling from vyos/vyos-build
00:00:12  Digest: sha256:912cae70d5ee1f652dd8dd1da940b646947256fbdd795a931f81bd9d9d4118e0
00:00:12  Status: Image is up to date for 172.17.17.17:5000/vyos/vyos-build:sagitta
00:00:12  172.17.17.17:5000/vyos/vyos-build:sagitta
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] isUnix
[Pipeline] withEnv
[Pipeline] {
[Pipeline] sh
00:00:13  + docker inspect -f . vyos/vyos-build:sagitta
00:00:13  .
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] withDockerContainer
00:00:13  Jenkins does not seem to be running inside a container
00:00:13  $ docker run -t -d -u 1006:1006 --sysctl net.ipv6.conf.lo.disable_ipv6=0 -e GOSU_UID=1006 -e GOSU_GID=1006 -w /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta -v /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta:/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta:rw,z -v /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta@tmp:/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta@tmp:rw,z -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** vyos/vyos-build:sagitta cat
00:00:13  $ docker top 1a8f5bbbe414fbd3cbb1ec659189ffcc68d5cc23c9bc9a16346957ab7ea0d16e -eo pid,comm
00:00:13  ERROR: The container started but didn't run the expected command. Please double check your ENTRYPOINT does execute the command passed as docker run argument, as required by official docker images (see https://github.com/docker-library/official-images#consistency for entrypoint consistency requirements).
00:00:13  Alternatively you can force image entrypoint to be disabled by adding option `--entrypoint=''`.
[Pipeline] {
[Pipeline] script
[Pipeline] {
[Pipeline] dir
00:00:13  Running in /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64
[Pipeline] {
[Pipeline] deleteDir
[Pipeline] checkout
00:00:13  Selected Git installation does not exist. Using Default
00:00:13  The recommended git tool is: NONE
00:00:13  No credentials specified
00:00:13  Warning: JENKINS-30600: special launcher org.jenkinsci.plugins.docker.workflow.WithContainerStep$Decorator$1@86550a; decorates hudson.Launcher$LocalLauncher@573d7367 will be ignored (a typical symptom is the Git executable not being run inside a designated container)
00:00:13  Cloning the remote Git repository
00:00:13  Cloning repository https://github.com/notvyos/vyos-xe-guest-utilities.git
00:00:13   > git init /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64 # timeout=10
00:00:13  Fetching upstream changes from https://github.com/notvyos/vyos-xe-guest-utilities.git
00:00:13   > git --version # timeout=10
00:00:13   > git --version # 'git version 2.39.5'
00:00:13   > git fetch --tags --force --progress -- https://github.com/notvyos/vyos-xe-guest-utilities.git +refs/heads/*:refs/remotes/origin/* # timeout=10
00:00:14   > git config remote.origin.url https://github.com/notvyos/vyos-xe-guest-utilities.git # timeout=10
00:00:14   > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
00:00:14  Avoid second fetch
00:00:14  Checking out Revision 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 (sagitta)
00:00:14   > git config core.sparsecheckout # timeout=10
00:00:14   > git checkout -f 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 # timeout=10
00:00:14  Commit message: "fixed jenkins library"
[Pipeline] sh
00:00:14  + git rev-parse --short=11 HEAD
[Pipeline] libraryResource
[Pipeline] sh
00:00:14  NOT_VYOS: 
[Pipeline] sh
00:00:15  + dpkg-buildpackage -b -us -uc -tc
00:00:15  dpkg-buildpackage: info: source package vyos-xe-guest-utilities
00:00:15  dpkg-buildpackage: info: source version 7.13.0+vyos1.3
00:00:15  dpkg-buildpackage: info: source distribution UNRELEASED
00:00:15  dpkg-buildpackage: info: source changed by Christian Poessinger <christian@poessinger.com>
00:00:15   dpkg-source --before-build .
00:00:15  dpkg-buildpackage: info: host architecture amd64
00:00:15   fakeroot debian/rules clean
00:00:15  dh clean --with systemd
00:00:15     dh_auto_clean
00:00:15  	make -j3 clean
00:00:15  make[1]: Entering directory '/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64'
00:00:15  rm -f -rf /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build
00:00:15  make[1]: Leaving directory '/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64'
00:00:15     dh_clean
00:00:15   debian/rules build
00:00:15  dh build --with systemd
00:00:15     dh_update_autotools_config
00:00:15     dh_autoreconf
00:00:15     dh_auto_configure
00:00:15     dh_auto_build
00:00:15  	make -j3
00:00:15  make[1]: Entering directory '/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64'
00:00:15  mkdir -p $(dirname /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./xe-daemon/xe-daemon.go)
00:00:15  mkdir -p $(dirname /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./syslog/syslog.go)
00:00:15  mkdir -p $(dirname /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./system/system.go)
00:00:15  cat /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/./xe-daemon/xe-daemon.go | \
00:00:15  sed -e "s/@PRODUCT_MAJOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MINOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MICRO_VERSION@/80/g" | \
00:00:15  sed -e "s/@NUMERIC_BUILD_NUMBER@/115/g" \
00:00:15  > /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./xe-daemon/xe-daemon.go
00:00:15  cat /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/./syslog/syslog.go | \
00:00:15  sed -e "s/@PRODUCT_MAJOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MINOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MICRO_VERSION@/80/g" | \
00:00:15  sed -e "s/@NUMERIC_BUILD_NUMBER@/115/g" \
00:00:15  > /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./syslog/syslog.go
00:00:15  cat /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/./system/system.go | \
00:00:15  sed -e "s/@PRODUCT_MAJOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MINOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MICRO_VERSION@/80/g" | \
00:00:15  sed -e "s/@NUMERIC_BUILD_NUMBER@/115/g" \
00:00:15  > /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./system/system.go
00:00:15  mkdir -p $(dirname /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./guestmetric/guestmetric.go)
00:00:15  mkdir -p $(dirname /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./guestmetric/guestmetric_linux.go)
00:00:15  cat /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/./guestmetric/guestmetric.go | \
00:00:15  sed -e "s/@PRODUCT_MAJOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MINOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MICRO_VERSION@/80/g" | \
00:00:15  sed -e "s/@NUMERIC_BUILD_NUMBER@/115/g" \
00:00:15  > /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./guestmetric/guestmetric.go
00:00:15  mkdir -p $(dirname /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./xenstoreclient/xenstore.go)
00:00:15  cat /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/./guestmetric/guestmetric_linux.go | \
00:00:15  sed -e "s/@PRODUCT_MAJOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MINOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MICRO_VERSION@/80/g" | \
00:00:15  sed -e "s/@NUMERIC_BUILD_NUMBER@/115/g" \
00:00:15  > /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./guestmetric/guestmetric_linux.go
00:00:15  mkdir -p $(dirname /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./xenstore/xenstore.go)
00:00:15  cat /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/./xenstoreclient/xenstore.go | \
00:00:15  sed -e "s/@PRODUCT_MAJOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MINOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MICRO_VERSION@/80/g" | \
00:00:15  sed -e "s/@NUMERIC_BUILD_NUMBER@/115/g" \
00:00:15  > /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./xenstoreclient/xenstore.go
00:00:15  cat /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/./xenstore/xenstore.go | \
00:00:15  sed -e "s/@PRODUCT_MAJOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MINOR_VERSION@/6/g" | \
00:00:15  sed -e "s/@PRODUCT_MICRO_VERSION@/80/g" | \
00:00:15  sed -e "s/@NUMERIC_BUILD_NUMBER@/115/g" \
00:00:15  > /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./xenstore/xenstore.go
00:00:15  mkdir -p /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/obj
00:00:15  mkdir -p /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/obj
00:00:15  go build -v -o /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/obj/xe-daemon /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./xe-daemon/xe-daemon.go
00:00:15  go build -v -o /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/obj/xenstore /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/./xenstore/xenstore.go
00:00:16  _/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/system
00:00:16  _/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/syslog
00:00:16  _/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/syslog
00:00:16  _/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/xenstoreclient
00:00:16  _/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/xenstoreclient
00:00:16  _/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/gobuild/guestmetric
00:00:16  command-line-arguments
00:00:16  command-line-arguments
00:00:16  ( mkdir -p /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/dist ; \
00:00:16    install -d /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/etc/init.d/ ; \
00:00:16    install -m 755 /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/mk/xe-linux-distribution.init /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/etc/init.d/xe-linux-distribution ; \
00:00:16    install -d /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/usr/sbin/ ; \
00:00:16    install -m 755 /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/mk/xe-linux-distribution /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/usr/sbin/xe-linux-distribution ; \
00:00:16    install -m 755 /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/obj/xe-daemon /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/usr/sbin/xe-daemon ; \
00:00:16    install -d /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/usr/bin/ ; \
00:00:16    install -m 755 /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/obj/xenstore /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/usr/bin/xenstore ; \
00:00:16    ln -sf xenstore /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/usr/bin/xenstore-read ; \
00:00:16    ln -sf xenstore /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/usr/bin/xenstore-write ; \
00:00:16    ln -sf xenstore /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/usr/bin/xenstore-exists ; \
00:00:16    ln -sf xenstore /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/usr/bin/xenstore-rm ; \
00:00:16    ln -sf xenstore /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/usr/bin/xenstore-list ; \
00:00:16    install -d /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/etc/udev/rules.d/ ; \
00:00:16    install -m 644 /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/mk/xen-vcpu-hotplug.rules /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage/etc/udev/rules.d/z10_xen-vcpu-hotplug.rules ; \
00:00:16    cd /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/stage ; \
00:00:16    tar cf /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build/dist/xe-guest-utilities_6.6.80-115_x86_64.tgz * \
00:00:16  )
00:00:16  make[1]: Leaving directory '/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64'
00:00:16     dh_auto_test
00:00:16     create-stamp debian/debhelper-build-stamp
00:00:16   fakeroot debian/rules binary
00:00:16  dh binary --with systemd
00:00:16     dh_testroot
00:00:16     dh_prep
00:00:16     dh_auto_install --destdir=debian/vyos-xe-guest-utilities/
00:00:17     dh_install
00:00:17     dh_installdocs
00:00:17     dh_installchangelogs
00:00:17     debian/rules override_dh_systemd_enable
00:00:17  make[1]: Entering directory '/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64'
00:00:17  dh_systemd_enable --name xe-guest-utilities.service
00:00:17  make[1]: Leaving directory '/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64'
00:00:17     debian/rules override_dh_installinit
00:00:17  make[1]: Entering directory '/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64'
00:00:17  dh_installinit --no-restart-on-upgrade --name xe-guest-utilities
00:00:17  dh_installinit: warning: --no-restart-on-upgrade has been renamed to --no-stop-on-upgrade
00:00:17  dh_installinit: warning: This feature will be removed in compat 12.
00:00:17  make[1]: Leaving directory '/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64'
00:00:17     dh_systemd_start
00:00:17     dh_perl
00:00:17     dh_link
00:00:17     dh_strip_nondeterminism
00:00:17     dh_compress
00:00:17     dh_fixperms
00:00:17     dh_missing
00:00:17     dh_strip
00:00:17  dh_strip: warning: Could not find the BuildID in debian/vyos-xe-guest-utilities/usr/sbin/xe-daemon
00:00:17     dh_makeshlibs
00:00:17     dh_shlibdeps
00:00:17     dh_installdeb
00:00:17     dh_gencontrol
00:00:17     dh_md5sums
00:00:17     dh_builddeb
00:00:17  dpkg-deb: building package 'vyos-xe-guest-utilities' in '../vyos-xe-guest-utilities_7.13.0+vyos1.3_amd64.deb'.
00:00:18   dpkg-genbuildinfo --build=binary -O../vyos-xe-guest-utilities_7.13.0+vyos1.3_amd64.buildinfo
00:00:18   dpkg-genchanges --build=binary -O../vyos-xe-guest-utilities_7.13.0+vyos1.3_amd64.changes
00:00:18  dpkg-genchanges: info: binary-only upload (no source code included)
00:00:18   fakeroot debian/rules clean
00:00:18  dh clean --with systemd
00:00:19     dh_auto_clean
00:00:19  	make -j3 clean
00:00:19  make[1]: Entering directory '/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64'
00:00:19  rm -f -rf /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64/build
00:00:19  make[1]: Leaving directory '/var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta/build-amd64'
00:00:19     dh_clean
00:00:19   dpkg-source --after-build .
00:00:19  dpkg-buildpackage: info: binary-only upload (no source included)
00:00:19  + rm -f mk/Makefile.deb
[Pipeline] }
[Pipeline] // dir
[Pipeline] archiveArtifacts
00:00:19  Archiving artifacts
00:00:19  Recording fingerprints
[Pipeline] archiveArtifacts
00:00:19  Archiving artifacts
00:00:19  ‘**/*.dsc’ doesn’t match anything: ‘**’ exists but not ‘**/*.dsc’
[Pipeline] echo
00:00:19  Archiving failed, ignoring - no source packages
[Pipeline] stash
00:00:19  Stashed 2 file(s)
[Pipeline] stash
[Pipeline] echo
00:00:19  Stashing failed, ignoring - no source packages
[Pipeline] }
[Pipeline] // script
Post stage
[Pipeline] deleteDir
[Pipeline] }
00:00:19  $ docker stop --time=1 1a8f5bbbe414fbd3cbb1ec659189ffcc68d5cc23c9bc9a16346957ab7ea0d16e
00:00:20  $ docker rm -f --volumes 1a8f5bbbe414fbd3cbb1ec659189ffcc68d5cc23c9bc9a16346957ab7ea0d16e
[Pipeline] // withDockerContainer
[Pipeline] }
[Pipeline] // withDockerRegistry
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // parallel
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Finalize)
[Pipeline] node
00:00:21  Running on [Jenkins](http://192.168.3.180:8080/computer/(built-in)/) in /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta
[Pipeline] {
[Pipeline] checkout
00:00:21  Selected Git installation does not exist. Using Default
00:00:21  The recommended git tool is: NONE
00:00:21  No credentials specified
00:00:21  Cloning the remote Git repository
00:00:21  Cloning repository https://github.com/notvyos/vyos-xe-guest-utilities.git
00:00:21   > git init /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta # timeout=10
00:00:21  Fetching upstream changes from https://github.com/notvyos/vyos-xe-guest-utilities.git
00:00:21   > git --version # timeout=10
00:00:21   > git --version # 'git version 2.39.5'
00:00:21   > git fetch --tags --force --progress -- https://github.com/notvyos/vyos-xe-guest-utilities.git +refs/heads/*:refs/remotes/origin/* # timeout=10
00:00:21   > git config remote.origin.url https://github.com/notvyos/vyos-xe-guest-utilities.git # timeout=10
00:00:21   > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
00:00:21  Avoid second fetch
00:00:21  Checking out Revision 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 (sagitta)
00:00:21   > git config core.sparsecheckout # timeout=10
00:00:21   > git checkout -f 695a766bf3c165b0cf9a04e6373b5d6a9e37b3b4 # timeout=10
00:00:21  Commit message: "fixed jenkins library"
[Pipeline] withEnv
[Pipeline] {
[Pipeline] echo
00:00:22  Warning, empty changelog. Have you run checkout?
[Pipeline] script
[Pipeline] {
[Pipeline] unstash
[Pipeline] unstash
[Pipeline] echo
00:00:22  Unstash failed, ignoring - could be because there exists no arm64 build
[Pipeline] unstash
[Pipeline] unstash
[Pipeline] echo
00:00:22  Unstash failed, ignoring - no source packages
[Pipeline] echo
00:00:22  Uploading Artifacts to external repository
[Pipeline] copyArtifacts
00:00:22  Copied 2 artifacts from "[vyos-xe-guest-utilities » sagitta](http://192.168.3.180:8080/job/vyos-xe-guest-utilities/job/sagitta/)" build number [2](http://192.168.3.180:8080/job/vyos-xe-guest-utilities/job/sagitta/2/)
[Pipeline] sshagent
00:00:22  [ssh-agent] Using credentials jenkins
00:00:22  $ ssh-agent
00:00:22  SSH_AUTH_SOCK=/tmp/ssh-XXXXXXLXoiUv/agent.214108
00:00:22  SSH_AGENT_PID=214111
00:00:22  Running ssh-add (command line suppressed)
00:00:22  Identity added: /var/lib/jenkins/workspace/vyos-xe-guest-utilities_sagitta@tmp/private_key_3917707052708190570.key (jenkins)
00:00:22  [ssh-agent] Started.
[Pipeline] {
[Pipeline] sh
00:00:22  + ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR jenkins@172.17.17.17 -t bash --login -c 'mkdir -p ~/VyOS/sagitta'
[Pipeline] findFiles
[Pipeline] findFiles
[Pipeline] findFiles
[Pipeline] echo
00:00:22  Remove deprecated binary package(s) from the repository...
[Pipeline] sh
00:00:23  + dpkg-deb -f mk/Makefile.deb Package
00:00:23  dpkg-deb: error: 'mk/Makefile.deb' is not a Debian format archive
[Pipeline] }
00:00:23  $ ssh-agent -k
00:00:23  unset SSH_AUTH_SOCK;
00:00:23  unset SSH_AGENT_PID;
00:00:23  echo Agent pid 214111 killed;
00:00:23  [ssh-agent] Stopped.
[Pipeline] // sshagent
[Pipeline] }
[Pipeline] // script
Post stage
[Pipeline] deleteDir
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] }
[Pipeline] // timeout
[Pipeline] End of Pipeline
ERROR: script returned exit code 2
Finished: FAILURE

 ```
</details>

**Comment by dd010101** [comment]
TL;DR
Just update the sagitta vyos-build-container by manually building it via Build now in the Jenkins. Then it will work again.

---

The vyos jenkins build was broken since the vyos-xe-guest-utilities contains extra deb file called "Makefile.deb" and this file wasn't excluded nor deleted, that's not actually a deb package, that's why the build fails because it tries to add Makefile.deb into apt repository, but that's invalid operation.

That's why I patched the vyos build container to exclude Makefile.deb.

There was change to use sagitta vyos build container - originally the `current` was used for many packages (incorrect leftover from development).

Yet the workaround was only in the `current` vyos build container, that's why I patched the sagitta vyos build container to include the same workaround. Your sagitta vyos build container is older that's why it doesn't have it. Update it and it will work again. Also make sure you are using the https://github.com/NOTvyos/vyos-build repository as the source for vyos-build-container if you still use the original dd010101/vyos-build repository.

**Comment by dutty5** [comment]
Actually, after the first failure I did exactly that. The new container was rebuilt in Jenkins successfully, however it did not solve the the issue. I think something else is cached in my case.

**Comment by dd010101** [comment]
Yes, the workaround is included in the Jenkins library from vyos-build not in the vyos-build-container. Thus go into JENKINS_URL/manage/configure and make sure the section **Global Trusted Pipeline Libraries** has the https://github.com/NOTvyos/vyos-build repository and not the older one.

**Comment by dutty5** [comment]
Changed. Do I have to rebuild the container after this change?

**Comment by dd010101** [comment]
No - this has nothing to do with the container, that was my mistake. Just build the failing package.

**Comment by dutty5** [comment]
Hello, I tried to build equuleus iso today, the build fails:

```
E: Failed to fetch http://repo.powerdns.com/debian/pool/main/p/pdns-recursor/pdns-recursor_4.8.9-1pdns.buster_amd64.deb  404  Not Found [IP: 188.166.116.224 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
P: Begin unmounting filesystems...
P: Saving caches...
Reading package lists...
Building dependency tree...
Reading state information...
make: *** [Makefile:32: iso] Error 1
ISO build failed
```

Looks like there is no buster version 4.8.9 of the pdns-recursor in this repo. 
How to fix this?

**Comment by dd010101** [comment]
Indeed, it's gone now. This is result of mistake or the buster support was dropped? I'm not sure... The package is still present in the http://repo.powerdns.com/debian/dists/buster-rec-48/main/binary-amd64/Packages so this seems like mistake - if the metadata is present yet the binary is missing - this doesn't looks like it was removed intentionally?

Did you see this issue earlier than today? Maybe it's just temporary?

**Comment by dutty5** [comment]
No, I didn't see this before. Can we find this binary somewhere else? Mirrors? Or build ? Or switch to another version?

**Comment by dd010101** [comment]
All those options are valid solutions - although neither is easy or ideal.

You could find the package somewhere, although I don't know where, you would need build where you still have temporary files and those are typically automatically deleted thus normally you wouldn't have them. I don't know about any mirrors and the mirror could likely mirror whatever the state is now, thus it may not be helpful.

We could totally [build](https://doc.powerdns.com/recursor/appendices/compiling.html) the recursor although I'm not how exactly sure you make .deb package from it, I didn't look into that.

Switching to another version would help only if you switched to package for different distribution - that is different build of the same version since currently there is no buster binary apart of `5.1.0~alpha1` - would it work to use bullseye binary? Maybe... The recursor has specific dependencies that may not align with buster:

Bullseye depends of recursor
```
Depends: adduser, dns-root-data, libboost-context1.74.0 (>= 1.74.0), libboost-filesystem1.74.0 (>= 1.74.0), libc6 (>= 2.29), libcap2 (>= 1:2.10), libcurl4 (>= 7.16.2), libfstrm0 (>= 0.4.0), libgcc-s1 (>= 3.4), libluajit-5.1-2 (>= 2.0.4+dfsg), libsnmp40 (>= 5.9+dfsg), libsodium23 (>= 0.6.1), libssl1.1 (>= 1.1.1), libstdc++6 (>= 9), libsystemd0
```

Buster depends of recursor
```
Depends: adduser, dns-root-data, libboost-context1.67.0, libboost-filesystem1.67.0, libboost-system1.67.0, libc6 (>= 2.27), libcap2 (>= 1:2.10), libcurl4 (>= 7.16.2), libfstrm0 (>= 0.4.0), libgcc1 (>= 1:3.4), libluajit-5.1-2 (>= 2.0.4+dfsg), libsnmp30 (>= 5.7.3+dfsg), libsodium23 (>= 0.6.1), libssl1.1 (>= 1.1.1), libstdc++6 (>= 7), libsystemd0
```

Thus the likely option to work would be 1) find the binary somewhere (depending on the availability, this would be temporary workaround not real solution) 2) build your own. If the buster support was indeed removed then we would need to implement the latter solution eventually.

**Comment by dd010101** [comment]
The pdns-recursor is still broken. I was thinking that the official repo would be fixed in short time but it wasn't. Thus I did create new [Jenkins job](https://github.com/dd010101/vyos-missing/blob/equuleus/packages/pdns-recursor/Jenkinsfile) to build PowerDNS Recursor 4.8.9 for equuleus (Debian buster), this should fix the equuleus build. You need to make sure you provision and build project jobs (step 6 and 7) in order to build the new pdns-recursor job, then it should work again.

**Comment by dutty5** [comment]
Thank you for your effort. I will try to build tomorrow or day after tomorrow. Yeah, I checked the pdns recursor package issue about a week ago. It was still broken then. To me it looks intentional, few other buster packages were also removed from that repo.

**Comment by dutty5** [comment]
Does anyone follow them? 😅 
Did they release Stream?

**Comment by dd010101** [comment]
> Interestingly, I found that few packages in equuleus branch got updates in January after at least of half a year of stall. No updates for the same packages in saggita branch. In a sense it looks like we can get newer and supported version of equuleus, but not sagitta or circinus which development is totally hidden.

The equuleus branch was updated for some packages? I checked the main ones (vyos-1x, vyos-build) and few recently updated and I didn't see any updates for equuleus. Could you give a example? I guess that needs to be a mistake more than anything.

> Did they release Stream?

Not yet.

**Comment by dutty5** [comment]
![Image](https://github.com/user-attachments/assets/6ab9d204-f969-4c19-bfa9-1f14fb6c68c4)

![Image](https://github.com/user-attachments/assets/6785f2a3-a7cd-46a8-b46b-1fac63060d35)

![Image](https://github.com/user-attachments/assets/d106d6ea-cea7-4626-abb4-959db1e32bb8)

![Image](https://github.com/user-attachments/assets/5176b42d-2270-4270-97c1-890a31aa8c6f)

These packages were automatically built on Jan 21 in my equuleus repo.

**Comment by dutty5** [comment]
May be related to your changes with regard to pdns-recursor, as it looks like you published your post around the same time

**Comment by dd010101** [comment]
Yep. Those aren't new packages. When I did add the pds-recursor then it seems like those changes did trigger rebuild of some other packages as side effect. If you look carefully then as example netfilter build script doesn't specify what filter should be used to filter only relevant commits and thus any commit will trigger rebuild https://github.com/notvyos/vyos-build/blob/equuleus/packages/netfilter/Jenkinsfile#L52. Those aren't updated packages - it's the same source code (6 months or older), it's just newer build.

**Comment by dutty5** [comment]
I see, no problem. Then at least the situation has not changed to the worse ) Which is anyway good

**Comment by dutty5** [comment]
Hi, looks like the Stream is released with the source code tarball.

https://blog.vyos.io/vyos-stream-1.5-2025-q1
https://vyos.net/get/stream/#1.5-2025-Q1

**Comment by dd010101** [comment]
Yep, I noticed few days ago https://github.com/dd010101/vyos-jenkins/discussions/56#discussioncomment-12263135

I plan to make a script to extract what they give us and commit the changes to corresponding repo but the contents are quite messy so it would take some work but looks like quite workable at first glance.

**Comment by dutty5** [comment]
Meanwhile, my working jenkins builds have stopped working. Containers do not build.

Today, as usually I updated the scripts:
```
rm -rf /opt/vyos-jenkins/
wget https://github.com/dd010101/vyos-jenkins/archive/refs/heads/master.tar.gz -O /tmp/vyos-jenkins.tar.gz
tar -xf /tmp/vyos-jenkins.tar.gz -C /tmp
mv /tmp/vyos-jenkins-master /opt/vyos-jenkins
cd /opt/vyos-jenkins
./1-prereqs.sh
./2-jenkins.sh
./3-repositories.sh
./4-uncron.sh
./5-docker-jobs.sh
```

And the 5th script has started the containers' rebuild. Both equuleus and sagitta containers have failed, with the nearly identical error:

<details>
  <summary>equuleus error</summary>

```
21:02:43  #19 [15/73] RUN curl https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh       --output /tmp/opam_install.sh --retry 10 --retry-delay 5 &&     sed -i 's/read -r BINDIR/BINDIR=""/' /tmp/opam_install.sh && sh /tmp/opam_install.sh &&     opam init --root=/opt/opam --comp=4.12.0 --disable-sandboxing
21:02:44  #19 0.189   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
21:02:44  #19 0.189                                  Dload  Upload   Total   Spent    Left  Speed
21:02:44  #19 0.190 
21:02:44    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
21:02:44  100 61474  100 61474    0     0   392k      0 --:--:-- --:--:-- --:--:--  394k
21:02:44  #19 0.353 ## Downloading opam 2.3.0 for linux on x86_64...
21:02:45  #19 1.780 ## Downloaded.
21:02:45  #19 1.780 ## Where should it be installed ? [/usr/local/bin] 
21:02:45  #19 ERROR: process "/bin/sh -c curl https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh       --output /tmp/opam_install.sh --retry 10 --retry-delay 5 &&     sed -i 's/read -r BINDIR/BINDIR=\"\"/' /tmp/opam_install.sh && sh /tmp/opam_install.sh &&     opam init --root=/opt/opam --comp=4.12.0 --disable-sandboxing" did not complete successfully: exit code: 1
21:02:45  ------
21:02:45   > [15/73] RUN curl https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh       --output /tmp/opam_install.sh --retry 10 --retry-delay 5 &&     sed -i 's/read -r BINDIR/BINDIR=""/' /tmp/opam_install.sh && sh /tmp/opam_install.sh &&     opam init --root=/opt/opam --comp=4.12.0 --disable-sandboxing:
21:02:45  1.780 ## Where should it be installed ? [/usr/local/bin] 
21:02:45  1.780 ## Where should it be installed ? [/usr/local/bin] 
21:02:45  0.189   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
21:02:45  0.189                                  Dload  Upload   Total   Spent    Left  Speed
21:02:45  
21:02:45  100 61474  100 61474    0     0   392k      0 --:--:-- --:--:-- --:--:--  394k
21:02:45  0.353 ## Downloading opam 2.3.0 for linux on x86_64...
21:02:45  1.780 ## Downloaded.
21:02:45  ------
21:02:45  Dockerfile:142
21:02:45  --------------------
21:02:45   141 |     # Installing OCAML needed to compile libvyosconfig
21:02:45   142 | >>> RUN curl https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh \
21:02:45   143 | >>>       --output /tmp/opam_install.sh --retry 10 --retry-delay 5 && \
21:02:45   144 | >>>     sed -i 's/read -r BINDIR/BINDIR=""/' /tmp/opam_install.sh && sh /tmp/opam_install.sh && \
21:02:45   145 | >>>     opam init --root=/opt/opam --comp=4.12.0 --disable-sandboxing
21:02:45   146 |     
21:02:45  --------------------
21:02:45  ERROR: failed to solve: process "/bin/sh -c curl https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh       --output /tmp/opam_install.sh --retry 10 --retry-delay 5 &&     sed -i 's/read -r BINDIR/BINDIR=\"\"/' /tmp/opam_install.sh && sh /tmp/opam_install.sh &&     opam init --root=/opt/opam --comp=4.12.0 --disable-sandboxing" did not complete successfully: exit code: 1
21:02:45  exit status 1
```
</details>

<details>
  <summary>sagitta error</summary>

```
21:02:28  #15 [11/45] RUN curl https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh       --output /tmp/opam_install.sh --retry 10 --retry-delay 5 &&     sed -i 's/read -r BINDIR/BINDIR=""/' /tmp/opam_install.sh && sh /tmp/opam_install.sh &&     opam init --root=/opt/opam --comp=4.14.2 --disable-sandboxing --no-setup
21:02:29  #15 0.184   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
21:02:29  #15 0.185                                  Dload  Upload   Total   Spent    Left  Speed
21:02:29  #15 0.185 
21:02:29    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
21:02:29    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
21:02:29  100 61474  100 61474    0     0   200k      0 --:--:-- --:--:-- --:--:--  200k
21:02:29  #15 0.494 ## Downloading opam 2.3.0 for linux on x86_64...
21:02:31  #15 2.083 ## Downloaded.
21:02:31  #15 2.083 ## Where should it be installed ? [/usr/local/bin] 
21:02:31  #15 ERROR: process "/bin/sh -c curl https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh       --output /tmp/opam_install.sh --retry 10 --retry-delay 5 &&     sed -i 's/read -r BINDIR/BINDIR=\"\"/' /tmp/opam_install.sh && sh /tmp/opam_install.sh &&     opam init --root=/opt/opam --comp=${OCAML_VERSION} --disable-sandboxing --no-setup" did not complete successfully: exit code: 1
21:02:31  ------
21:02:31   > [11/45] RUN curl https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh       --output /tmp/opam_install.sh --retry 10 --retry-delay 5 &&     sed -i 's/read -r BINDIR/BINDIR=""/' /tmp/opam_install.sh && sh /tmp/opam_install.sh &&     opam init --root=/opt/opam --comp=4.14.2 --disable-sandboxing --no-setup:
21:02:31  2.083 ## Where should it be installed ? [/usr/local/bin] 
21:02:31  
21:02:31  
21:02:31  100 61474  100 61474    0     0   200k      0 --:--:-- --:--:-- --:--:--  200k
21:02:31  0.494 ## Downloading opam 2.3.0 for linux on x86_64...
21:02:31  ------
21:02:31  Dockerfile:133
21:02:31  --------------------
21:02:31   132 |     # Installing OCAML needed to compile libvyosconfig
21:02:31   133 | >>> RUN curl https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh \
21:02:31   134 | >>>       --output /tmp/opam_install.sh --retry 10 --retry-delay 5 && \
21:02:31   135 | >>>     sed -i 's/read -r BINDIR/BINDIR=""/' /tmp/opam_install.sh && sh /tmp/opam_install.sh && \
21:02:31   136 | >>>     opam init --root=/opt/opam --comp=${OCAML_VERSION} --disable-sandboxing --no-setup
21:02:31   137 |     
21:02:31  --------------------
21:02:31  ERROR: failed to solve: process "/bin/sh -c curl https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh       --output /tmp/opam_install.sh --retry 10 --retry-delay 5 &&     sed -i 's/read -r BINDIR/BINDIR=\"\"/' /tmp/opam_install.sh && sh /tmp/opam_install.sh &&     opam init --root=/opt/opam --comp=${OCAML_VERSION} --disable-sandboxing --no-setup" did not complete successfully: exit code: 1
21:02:31  exit status 1
```
</details>

Restarting the builds of the containers from jenkins manually did not help, same error.

I feel uncomfortable distracting you from the probably more important work to figure out the new build process with regard to the Stream, however if you have ideas what's wrong here and how it can be fixed it would be very helpful. I still rebuild both equuleus and sagitta from time to time to have fresh ISOs in case some OS packages got updates. Thanks.

**Comment by dd010101** [comment]
This would be it https://github.com/vyos/vyos-build/commit/841b0f41724a5d6478e21865189679cf75cdee4b, backported to both equuleus and sagitta. You can try now.

**Comment by dutty5** [comment]
Yes, now it works. Thank you.
Interestingly, a number of packages got rebuilds, including nftables for both equuleus and sagitta, and the kernel for equuleus.

**Comment by dd010101** [comment]
This is known side effect of making changes in the vyos-build repository, where some packages have less than ideal definition, and they are specified to be rebuild on any changes in the whole repo - thus the rebuild is triggered by irrelevant change - like in this example where the docker for build environment was changed.
